### PR TITLE
feat(hooks): added useCurrentUser hook

### PR DESCRIPTION
**Pull request by jordan-a-young**
This hook fetches the currently logged in user

### CODE REVIEW

**Comment by robmcguinness** [comment]
curious: what would be the value in this over just calling the sdk directly?

File: packages/hooks/src/useCurrentUser.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from 'react';
```
Comment (nylon22) [inline packages/hooks/src/useCurrentUser.js:1]: Please add to README

File: packages/hooks/src/useCurrentUser.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  const [error, setError] = useState();
+
+  useEffect(() => {
+    let ignore = false;
+
+    const fetchUser = async () => {
+      setLoading(true);
+      try {
+        const response = await avUserApi.me();
+        if (!ignore) setUser(response);
```
Comment (nylon22) [inline packages/hooks/src/useCurrentUser.js:16]: Can you explain use case for `ignore` arg?

**Comment by jordan-a-young** [comment]
Hooks make it easier to manage if an async call is loading, in my opinion.
That was my main motivation here.

**Comment by jordan-a-young** [comment]
This is something I copied from Dan Abramov. The use case is, what if the
component unmounts before the API call finishes? This keeps us from calling
setState on an unmounted component.

**Comment by robmcguinness** [comment]
@jordan-a-young appreciate feedback 🙌

File: packages/hooks/src/useCurrentUser.js (hunk 1 above)
Comment (jordan-a-young) [reply packages/hooks/src/useCurrentUser.js:1]: I added an example to the README as well as a story for Storybook

File: packages/hooks/README.md [hunk 3]
```diff
@@ -155,9 +155,29 @@ import React, { useState } from 'react';
...
-  return <div>Test Component</div>;
+  return <div>{loading ? 'Loading...' : region.value}</div>;
+};
+// ...
+```
+
+
+#### useCurrentUser
+
+Hook that will return the current user's region.
```
Comment (nylon22) [inline packages/hooks/README.md]: Copy pasta 😉

Comment (jordan-a-young) [reply packages/hooks/README.md]: I thought I got em all....

**Comment by GoPro16** [comment]
Going to use this PR as guinea pig for new CICD since it doesn't need to be merged immediately.

**Pull request merged**